src/cluster/registration.go

```diff
@@ -51,22 +53,52 @@ func (c DeviceCluster) HandleDiscoveredDevice(newDevice model.NewDevice) error {
 		return err
 	}
 
-	if len(results) > 0 {
-		return nil
-	}
-
-	completeDevice, err := c.DeviceRepository.Create(newDevice)
-	if err != nil {
-		return err
+	var discoveredDevice *model.Device
+	switch len(results) {
+	case 0:
+		discoveredDevice, err = c.DeviceRepository.Create(newDevice)
+		if err != nil {
+			return err
+		}
+	case 1:
+		discoveredDevice = updateDeviceWithDiscoveredData(results[0], newDevice)
+		discoveredDevice.Active = true
+		err := c.DeviceRepository.Update(updateDeviceFromDevice(discoveredDevice))
+		if err != nil {
+			return err
+		}
+	default:
+		return errors.New("multiple results returned for 1 mac address")
 	}
 
-	c.logger.Debug(fmt.Sprintf("registered mac address [%s] with id [%s]", completeDevice.MAC, completeDevice.ID))
+	c.logger.Debug(fmt.Sprintf("registered mac address [%s] with id [%s] at [%s]:[%s]", discoveredDevice.MAC, discoveredDevice.ID, newDevice.Host, strconv.Itoa(newDevice.Port)))
 
-	deviceWrapper := device.NewDeviceWrapper(*completeDevice)
-	err = deviceWrapper.RunHealthCheck(c.DeviceClient)
-	if err != nil {
+	// `Immediately` run health check
+	if err := device.NewDeviceWrapper(*discoveredDevice).RunHealthCheck(c.DeviceClient); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// updateDeviceWithDiscoveredData updates fields in a device that may change between discoveries.
+func updateDeviceWithDiscoveredData(dev *model.Device, discovered model.NewDevice) *model.Device {
```

Comment:
Is there a better place to put these? I was thinking of moving them to be methods on `model.Device`.